lib/matplotlib/lines.py

```diff
@@ -772,6 +774,16 @@ def draw(self, renderer):
                 if self.get_sketch_params() is not None:
                     gc.set_sketch_params(*self.get_sketch_params())
 
+                # We first draw the path below if needed.
+                if self.is_dashed() and self._offcolor is not None:
+                    lc_rgba = mcolors.to_rgba(self._offcolor, self._alpha)
+                    gc.set_foreground(lc_rgba, isRGBA=True)
+                    gc.set_dashes(0, None)
```

Comment:
I don't think this is enough. This will just draw a solid line in the background and doesn't look good if you set an alpha on the line. I think you'll need to unwrap the dash pattern and do the inverse dashed pattern.

Reply:
Good point! It also strikes me that the dash cap style can cause some issues here. But I guess that one can probably live with that having a round dash cap style with alpha gives a less than ideal result?

The option would be, I guess, to introduce an "inverted" round cap style. Also, projecting caps will require a bit of extra work here, but should be more easily doable since one knows the additional overlap (linewidth/2).

Reply:
So my initial attempt at this failed badly because I assumed that `offset` would add an extra "gap" to the start of the sequence, when in fact it seems to remove a section from the start instead.  I couldn't find this documented anywhere: [this docstring](https://matplotlib.org/stable/api/backend_bases_api.html#matplotlib.backend_bases.GraphicsContextBase.set_dashes) links to a "blue book" that might help, but that pdf seems to be only available on certain high security systems.

I've sanity checked this by adapting the linestyle demo, plotting with and without an initial offset of 1, so I know it works for a variety of dash styles:

![linestyles](https://user-images.githubusercontent.com/10599679/172542192-59e9fd4a-74a7-4a4c-8c87-59bdc5d6b612.png)

![linestyles2](https://user-images.githubusercontent.com/10599679/172542217-8049b2ef-bb99-494c-b42b-50a845135870.png)

Reply:
It also turned out that the line underneath was problematic even without setting the alpha: the previous test image showed outlines around the lines which disappeared when I updated the main code:

![striped_line-failed-diff](https://user-images.githubusercontent.com/10599679/172543255-912e5fcf-ff8f-4950-bba5-a0280d659288.png)

Reply:
> I couldn't find this documented anywhere

I was under the same impression as you. That is clearly something that can be documented more clearly. Although now that I know it, I can sort of understand what `offset` means, and explain this line in the code:
https://github.com/matplotlib/matplotlib/blob/8e2987b9804b6eaa32b8ffa40059e0d5267960ee/lib/matplotlib/lines.py#L57

I opened #23220 to track this.

Reply:
> I couldn't find this documented anywhere: [this docstring](https://matplotlib.org/stable/api/backend_bases_api.html#matplotlib.backend_bases.GraphicsContextBase.set_dashes) links to a "blue book" that might help, but that pdf seems to be only available on certain high security systems.

https://www.adobe.com/jp/print/postscript/pdfs/PLRM.pdf page 667 
> The offset operand can be thought of as the “phase” of the dash pattern relative to
the start of the path. It is interpreted as a distance into the dash pattern (measured
in user space units) at which to start the pattern. Before beginning to stroke a
path, the stroke operator cycles through the elements of array, adding up dis-
tances and alternating dashes and gaps as usual, but without generating any out-
put. When the accumulated distance reaches the value specified by offset, it begins
stroking from the starting point of the path, using the dash pattern from the point
that has been reached. Each subpath of a path is treated independently; the dash
pattern is restarted and the offset reapplied at the beginning of each subpath.

and I'll open a PR to change the link 'cause this is at least not locked down